src/Database/Redis/URL.hs

```diff
@@ -0,0 +1,56 @@
+module Database.Redis.URL
+    ( parseConnectInfo
+    ) where
+
+import Control.Error.Util (note)
+import Control.Monad (guard)
+import Control.Monad.Plus (partial)
+import Data.Maybe (fromMaybe)
+import Database.Redis.Core (ConnectInfo(..), defaultConnectInfo)
+import Database.Redis.ProtocolPipelining (PortID(..))
+import Network.HTTP.Base
+import Network.URI (parseURI, uriPath, uriScheme)
+import Text.Read (readMaybe)
+
+import qualified Data.ByteString.Char8 as C8
+
+-- | Parse a @'ConnectInfo'@ from a URL
+--
+-- Username is ignored, path is used to specify the database:
+--
+-- >>> parseConnectInfo "redis://username:password@host:42/2"
+-- Right (ConnInfo {connectHost = "host", connectPort = PortNumber 42, connectAuth = Just "password", connectDatabase = 2, connectMaxConnections = 50, connectMaxIdleTime = 30s, connectTimeout = Nothing})
+--
+-- N.B. invalid (non-integer) databases are ignored and revert to default:
+--
+-- >>> connectDatabase <$> parseConnectInfo "redis://username:password@host:42/db"
+-- Right 0
+--
+-- The scheme is validated, to prevent mixing up configurations:
+--
+-- >>> parseConnectInfo "postgres://"
+-- Left "Wrong scheme"
+--
+-- Beyond that, all values are optional. Omitted values are taken from
+-- @'defaultConnectInfo'@:
+--
+-- >>> parseConnectInfo "redis://"
+-- Right (ConnInfo {connectHost = "localhost", connectPort = PortNumber 6379, connectAuth = Nothing, connectDatabase = 0, connectMaxConnections = 50, connectMaxIdleTime = 30s, connectTimeout = Nothing})
+--
+parseConnectInfo :: String -> Either String ConnectInfo
+parseConnectInfo url = do
+    uri <- note "Invalid URI" $ parseURI url
+    note "Wrong scheme" $ guard $ uriScheme uri == "redis:"
+    uriAuth <- note "Missing or invalid Authority"
+        $ parseURIAuthority
+        $ uriToAuthorityString uri
+
+    return defaultConnectInfo
+        { connectHost = fromMaybe (connectHost defaultConnectInfo)
```

Comment:
So you convert a bool to a maybe just to unwrap it back? Why not just `let h = host uriAuth in if null h then connectHost defaultConnectInfo else h`? Seems like we could get rid of a library dependency and conversions to/from Maybe value